Dalamud/Game/Addon/Events/AddonEventManager.cs

```diff
@@ -146,11 +156,34 @@ internal void RemovePluginEventController(string pluginId)
         if (this.pluginEventControllers.FirstOrDefault(entry => entry.PluginId == pluginId) is { } controller)
         {
             Log.Verbose($"Removing PluginEventController for: {pluginId}");
-            this.pluginEventControllers.Remove(controller);
-            controller.Dispose();
+            this.oldEventControllers.Add(controller);
         }
     }
+    
+    private void OnFrameworkUpdate(IFramework framework1)
+    {
+        if (this.newEventControllers.Any())
+        {
+            foreach (var toAdd in this.newEventControllers)
+            {
+                this.pluginEventControllers.Add(toAdd);
+            }
+        
+            this.newEventControllers.Clear();  
```

Comment:
Use [`TryTake`](https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentbag-1.trytake?view=net-8.0) instead; with the current construction of `foreach (bag) {...} /* A */ bag.Clear();`, if a plugin tries to add an event listener at a truly unfortunate timing of `/* A */`, its request would be ignored. 

```cs
while (this.newEventControllers.TryTake(out var toAdd))
{
    if (this.pluginEventControllers.All(x => x.PluginId != toAdd.PluginId))
        this.pluginEventControllers.Add(toAdd);
}
```